patches/kaggle_gcp.py

```diff
@@ -200,9 +200,70 @@ def monkeypatch_gcs(self, *args, **kwargs):
         storage.Client.__init__ = monkeypatch_gcs
     return storage
 
+def init_automl():
+    is_user_secrets_token_set = "KAGGLE_USER_SECRETS_TOKEN" in os.environ
+    from google.cloud import automl_v1beta1 as automl
+    if not is_user_secrets_token_set:
+        return automl
+
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_automl():
+        return automl
+
+    from kaggle_secrets import GcpTarget
+    from kaggle_gcp import KaggleKernelCredentials
+    kaggle_kernel_credentials = KaggleKernelCredentials(target=GcpTarget.AUTOML)
+
+    # The AutoML client library exposes 4 different client classes (AutoMlClient,
+    # TablesClient, PredictionServiceClient and GcsClient), so patch each of them.
+    # The same KaggleKernelCredentials are passed to all of them.
+
+    automl_client_init = automl.AutoMlClient.__init__
+    def monkeypatch_automl(self, *args, **kwargs):
+        specified_credentials = kwargs.get('credentials')
+        if specified_credentials is None:
+            Log.info("No credentials specified, using KaggleKernelCredentials.")
+            kwargs['credentials'] = kaggle_kernel_credentials
+        # Note: This is only here so that unit tests can check whether
+        # credentials were set properly.
+        self._kaggle_credentials = kwargs['credentials']
+        return automl_client_init(self, *args, **kwargs)
+
+    if (not has_been_monkeypatched(automl.AutoMlClient.__init__)):
+        automl.AutoMlClient.__init__ = monkeypatch_automl
+
+
+    automl_tablesclient_init = automl.TablesClient.__init__
+    def monkeypatch_tablesclient(self, *args, **kwargs):
+        specified_credentials = kwargs.get('credentials')
+        if specified_credentials is None:
+            Log.info("No credentials specified, using KaggleKernelCredentials.")
+            kwargs['credentials'] = kaggle_kernel_credentials
+        self._kaggle_credentials = kwargs['credentials']
+        return automl_tablesclient_init(self, *args, **kwargs)
+
+    if (not has_been_monkeypatched(automl.TablesClient.__init__)):
+        automl.TablesClient.__init__ = monkeypatch_tablesclient
```

Comment:
Totally a nit, and no change needed, but may make sense to do this monkeypatch_* as as higher level function so we don't need to duplicate that so many times?

Reply:
Done. Good idea! It cleaned up a lot of code.

Reply:
Nice 🙌